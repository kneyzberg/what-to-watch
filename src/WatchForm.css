.WatchForm-form {
  padding-left: 20%;
  padding-right: 20%;
  

}

.WatchForm-categories {
  color: #e89110;
  font-weight: bold;
  font-size: 1.2rem;
}