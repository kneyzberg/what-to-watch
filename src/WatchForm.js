import React, { useState } from "react";
import { decadeOptions, decadeMap } from "./decadeData";
import "./WatchForm.css";

function WatchForm({ genres, getMovies, streamers }) {
  console.log(streamers);

  const initialFormData = { genre: "", decade: "" };

  const [formData, setFormData] = useState(initialFormData);

  function handleChangeGenre(e) {
    console.log(e, "event");
    const { name, value } = e.target;
    setFormData((f) => ({ ...f, [name]: value }));
    console.log(formData, "form data");
  }

  function handleChangeDecade(e) {
    const { name, value } = e.target;
    console.log("value", value);

    const start = decadeMap[value].start;
    const end = decadeMap[value].end;
    setFormData((f) => ({
      ...f,
      start_date: start,
      end_date: end,
      [name]: value,
    }));
  }

  function handleCheckBox(e) {
    const { name, checked } = e.target;
    setFormData((f) => ({ ...f, [name]: checked }));
  }

  function handleSubmit(e) {
    e.preventDefault();
    getMovies(formData);
  }

  console.log(formData, "form data");

  return (
    <div>
      <h3>
        Fill out the form below to get movie reccomendations based on your mood!
      </h3>
      <form>
        <div>
          Genre:
          <select
            className="form-select m-3"
            onChange={handleChangeGenre}
            name="genre"
            id="genre"
            data-testid="genre"
          >
            <option value=""> Please pick a genre</option>
            {genres.map((g) => (
              <option value={g.id}>{g.name}</option>
            ))}
          </select>
        </div>
        <div>
          {" "}
          Decade:
          <select onChange={handleChangeDecade} name="decade" id="decade">
            <option value=""> Please pick a decade</option>
            {decadeOptions.map((d) => (
              <option value={d}>{d}</option>
            ))}
          </select>
        </div>
        {/* <div>
          Provider options:
          <ul>
            {streamers.map((s) => (
              <li>
                <input
                  onChange={handleCheckBox}
                  type="checkbox"
                  name={s.provider_id}
                  id={s.provider_id}
                ></input>
                <label htmlFor={s.provider_id}>{s.provider_name}</label>
              </li>
            ))}
          </ul>
        </div> */}
        <div>
          <button type="submit" onClick={handleSubmit}>
            submit
          </button>
        </div>
      </form>
    </div>
  );
}

export default WatchForm;
