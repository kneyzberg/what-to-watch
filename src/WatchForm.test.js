import WatchForm from "./WatchForm";
import React from "react";
import { render, fireEvent } from "@testing-library/react";

const genres = [
  {
    id: 1,
    name: "Comedy",
  },
  {
    id: 2,
    name: "Drama",
  },
  {
    id: 3,
    name: "Horror",
  },
];

describe("Testing the WatchForm component", () => {
  test("Watch Form renders correctly", () => {
    render(<WatchForm genres={genres} />);
  });
  test("matches snapshot", () => {
    const { container } = render(<WatchForm genres={genres} />);
    expect(container).toMatchSnapshot();
  });
  test("Picking a genre successfully", () => {
    const { getByTestId, getByText } = render(<WatchForm genres={genres} />);
    const selectElement = getByTestId("genre");
    fireEvent.change(selectElement, { target: { value: "2" } });
    expect(selectElement.value).toBe("2");
    expect(getByText("Drama")).toBeInTheDocument();
  });
});
