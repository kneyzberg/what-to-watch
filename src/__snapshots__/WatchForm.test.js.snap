// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the WatchForm component matches snapshot 1`] = `
<div>
  <div>
    <h3>
      Fill out the form below to get movie recommendations based on your mood!
    </h3>
    <form
      class="WatchForm-form"
    >
      <div>
        <span
          class="WatchForm-categories"
        >
          Genre:
        </span>
        <select
          class="form-select m-2"
          data-testid="genre"
          id="genre"
          name="genre"
        >
          <option
            value=""
          >
             Please pick a genre
          </option>
          <option
            value="1"
          >
            Comedy
          </option>
          <option
            value="2"
          >
            Drama
          </option>
          <option
            value="3"
          >
            Horror
          </option>
        </select>
      </div>
      <div>
         
        <span
          class="WatchForm-categories"
        >
          Decade:
        </span>
        <select
          class="form-select m-2"
          id="decade"
          name="decade"
        >
          <option
            value=""
          >
             Please pick a decade
          </option>
          <option
            value="1920s"
          >
            1920s
          </option>
          <option
            value="1930s"
          >
            1930s
          </option>
          <option
            value="1940s"
          >
            1940s
          </option>
          <option
            value="1950s"
          >
            1950s
          </option>
          <option
            value="1960s"
          >
            1960s
          </option>
          <option
            value="1970s"
          >
            1970s
          </option>
          <option
            value="1980s"
          >
            1980s
          </option>
          <option
            value="1990s"
          >
            1990s
          </option>
          <option
            value="2000s"
          >
            2000s
          </option>
          <option
            value="2010s"
          >
            2010s
          </option>
          <option
            value="2020s"
          >
            2020s
          </option>
        </select>
      </div>
      <div>
        <button
          class="btn btn-info mt-2"
          type="submit"
        >
          submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
